lib/jobs/secure-erase-drive.js

```diff
@@ -0,0 +1,386 @@
+        }
+    ]
+    eraseSettings example:
+    [
+        {
+            'disks': ['sda', 'sdb'],
+            'tool': 'hdparm',
+            'operation': 'secure-erase'
+        },
+        {
+            'disks': ['1'],
+            'tool': 'sg_format'
+        },
+        {
+            'disks': ['sdc']
+        }
+    ]
+
+    The output parameter will be:
+    [
+        {
+            "disks": [
+                {
+                    "diskName": "/dev/sda",
+                    "vd": "/c0/v0",
+                    "scsiId": "0:2:0:0",
+                    "did": [23],
+                    "sid": ["/c0/e36/s0"]
+                },
+                ...
+            ],
+            "tool": "hdparm",
+            "operation": "secure-erase",
+            "vendor": "lsi"
+        }
+        ...
+    ]
+    */
+    SecureEraseJob.prototype._marshalParams = function(diskInfo) {
+        var self = this;
+
+        return _.transform(self.eraseSettings, function(result, entry) {
+            /* entry ex:
+            {
+                'disks': ['sda', 'sdb'],
+                'tool': 'hdparm',
+                'operation': 'secure-erase'
+            }
+            */
+
+            _.forEach(entry.disks, function(disk, idx) {
+            // disk ex: 'sda'
+
+                var match = false;
+
+                _.forEach(diskInfo, function(info) {
+
+
+                    // Find out the matched entry from catalogSearch
+                    if ((info.devName === disk) ||
+                        (info.identifier.toString() === disk)) {
+
+                        match = true;
+
+                        // Verify the specified tool supports the disk's protocol
+                        if (entry.hasOwnProperty('tool')) {
+
+                            // Get the disk protocol
+                            var protocol = null;
+                            if (info.esxiWwid.indexOf('SATADOM') > 0) {
+                                protocol = 'SATADOM';
+                            }
+                            else if (info.hasOwnProperty('disks')) {
+                                // Disks under the same virtual disk uses the same protocol,
+                                // so check the first item is sufficient
+                                protocol = info.disks[0].protocol;
+                            }
+
+                            if (!_.contains(Constants.SecureErase[entry.tool].protocol, protocol)) {
+                                throw new Error(entry.tool + " doesn't support disk " + disk +
+                                                ' (' + protocol + ')');
+                            }
+                        }
+                        else {
+                            entry.tool = 'scrub';
+                        }
+
+                        // Marshal the disk parameter
+                        entry.disks[idx] = {
+                            'diskName': '/dev/' + info.devName,
```

Comment:
I think the secure_erase.py would add '/dev/' prefix

https://github.com/RackHD/on-http/pull/306/files#diff-6bb94a11f57ddfb92f2c9430d40160e3R108